cost-analyzer/templates/cost-analyzer-ingress-template.yaml

```
@@ -5,7 +5,7 @@
 {{- $ingressPaths := .Values.ingress.paths -}}
 {{- $ingressPathType := .Values.ingress.pathType -}}
 {{- $apiV1 := false -}}
-{{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1" }}
+{{- if and (.Capabilities.APIVersions.Has "networking.k8s.io/v1") (semverCompare "^1.19-0" .Capabilities.KubeVersion.GitVersion) }}
```

The [Helm docs](https://helm.sh/docs/chart_template_guide/builtin_objects/) say the following:

"`Capabilities.APIVersions.Has $version` indicates whether a version (e.g., `batch/v1`) or resource (e.g., `apps/v1/Deployment`) is available on the cluster."

Because of the `e.g., apps/v1/Deployment`, I think this will work:

```suggestion
{{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" }}
```

---

Another reference: https://austindewey.medium.com/how-to-check-that-a-kubernetes-api-resource-exists-when-using-helm-523d9791ba91

---

I used `helm install --dry-run` to test the change:

```
helm install kubecost ./cost-analyzer --namespace kubecost --set ingress.enabled=true --dry-run | grep "Ingress" -B 2 -A 15
W0131 15:56:26.196687  631098 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
W0131 15:56:26.211076  631098 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
# Source: cost-analyzer/templates/cost-analyzer-ingress-template.yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: kubecost-cost-analyzer
  labels:

    app.kubernetes.io/name: cost-analyzer
    helm.sh/chart: cost-analyzer-1.90.0-rc.0
    app.kubernetes.io/instance: kubecost
    app.kubernetes.io/managed-by: Helm
    app: cost-analyzer
spec:
  rules:
    - host: "cost-analyzer.local"
      http:
        paths:
          - path: /
```

---

Confirmed this works. Much cleaner, thank you!